### PR TITLE
Fix accessibility issues with payment request

### PR DESCRIPTION
**Pull request by Gubbsy**
## Proposed changes

Fix accessibility issues with payment request journey

### CODE REVIEW

File: app/views/partials/funding_application/progress_and_spend/_add_edit_high_spend.html.erb [hunk 1]
```diff
@@ -151,12 +151,6 @@
     <div class="govuk-date-input <%= "#{'govuk-form-group--error' if
       @high_spend.errors.any? { | er | [:date_day, :date_month, :date_year].include? er.attribute } }" %>" id="full-date">
 
-      <%=
-        f.label :description,
-        t('progress_and_spend.payments.high_spend_add_edit.date_of_spend'),
-        class: "govuk-heading-s govuk-!-margin-top-4"
-      %>
-
```
Comment (ptrelease) [inline app/views/partials/funding_application/progress_and_spend/_add_edit_high_spend.html.erb:159]: Why have you removed this label?  Required by design.

File: app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb [hunk 2]
```diff
@@ -63,17 +63,16 @@
           <%=
             f.check_box :higher_spend,
               {
-                  id: :select_higher_spend,
                   class: "govuk-checkboxes__input"
               },
               'true', 'false'
           %>
```
Comment (ptrelease) [inline app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb:69]: Links still work fine.  But may want to remove reference to select_higher_spend above (line 18).

File: app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb [hunk 3]
```diff
@@ -103,16 +101,15 @@
...
 
           <%=
             f.label :no_update,
-                  t('progress_and_spend.payments.what_spend.no_spend_update'),
-                  class: "govuk-label govuk-checkboxes__label"
+            t('progress_and_spend.payments.what_spend.no_spend_update'),
+            class: "govuk-label govuk-checkboxes__label"
           %>
 
         </div>
```
Comment (ptrelease) [inline app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb:115]: These changes work, but when a grantee returns, the tables show.

Although my empty table headers work fine with Voiceover, Wave has picked them up as errors.

I've figured a fox out - will push on this branch shortly.

Comment (ptrelease) [reply app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb:115]: Done.

File: app/views/partials/funding_application/progress_and_spend/_high_spend_summary_table.html.erb [hunk 4]
```diff
@@ -121,10 +120,18 @@
...
+
+              <th scope="col" class="govuk-table__header" style="visibility: hidden;">
+                <%= t('generic.no_text') %>
+              </th>
 
-            <th scope="col" class="govuk-table__header"/>
-            <th scope="col" class="govuk-table__header"/>
+            <% end %>
 
           </tr>
```
Comment (ptrelease) [inline app/views/partials/funding_application/progress_and_spend/_high_spend_summary_table.html.erb:137]: Will drop a note to designer about accessibility issue with red text.

File: app/views/partials/funding_application/progress_and_spend/_add_edit_high_spend.html.erb (hunk 1 above)
Comment (Gubbsy) [reply app/views/partials/funding_application/progress_and_spend/_add_edit_high_spend.html.erb:159]: Replaced with p tag as should not be an orphaned label

File: app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb (hunk 2 above)
Comment (Gubbsy) [reply app/views/funding_application/progress_and_spend/payments/what_spend/show.html.erb:69]: Fixed

File: app/views/partials/funding_application/progress_and_spend/_high_spend_summary_table.html.erb (hunk 4 above)
Comment (Gubbsy) [reply app/views/partials/funding_application/progress_and_spend/_high_spend_summary_table.html.erb:137]: 👍

**Review by ptrelease — APPROVED** [review]

**Pull request merged**